lib/enum/arrays.rb

```diff
@@ -1,24 +1,33 @@
 class Array
 
   def sum_of_numeric_elements
+    (self.group_by(&:class)[Fixnum] + self.group_by(&:class)[Float]).inject(:+)
   end
 
   def product_of_numeric_elements
+    (self.group_by(&:class)[Fixnum] + self.group_by(&:class)[Float]).inject(:*)
```

Comment:
Same comment here. Pretty much everywhere you are using group_by, it would be simpler and easier to use select.